### PR TITLE
[MODULAR] Adds a visible timer to CI that shows you how long you have til the timeout period expires

### PR DESCRIPTION
**Pull request by nikothedude**
## About The Pull Request

Title. Its a bit janky, but this was a hasty PR.

## How This Contributes To The Skyrat Roleplay Experience

CI is never ever used properly. THis might help.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  
https://github.com/Skyrat-SS13/Skyrat-tg/assets/59709059/de5be905-fb0d-4722-abae-097749b15b0f

</details>

## Changelog

:cl:
qol: CI now shows how long you have til you can engage in mechanics if the other party doesnt do anything
/:cl:

### CODE REVIEW

**Comment by nikothedude** [comment]
drafted til policy

**Comment by nikothedude** [comment]
and do note, the lag in that video is likely caused by this being locally hosted and at roundstart when things are already a ltitle sluggish

**Comment by nikothedude** [comment]
i have no fuuucking idea why ci is fialing and im pissed

**Comment by lessthnthree** [comment]
> i have no fuuucking idea why ci is fialing and im pissed

did you wait 7 seconds

**Comment by nikothedude** [comment]
> > i have no fuuucking idea why ci is fialing and im pissed
> 
> did you wait 7 seconds

im coming for you.

**Comment by vinylspiders** [comment]
> > i have no fuuucking idea why ci is fialing and im pissed
> 
> did you wait 7 seconds

lol

**Comment by vinylspiders** [comment]
Rather than having floating numbers which look somewhat jarring to me, I think a better looking way to do this would be based on the time that's elapsed have the CI icon change. 

You could have the fill of it gradually go away or lighten in color noticeably. Like 4 or 8 different icon states.

**Comment by nikothedude** [comment]
id argue its important to maintain the granularity of being able to see the exact numbers til the timeout period expires considering how imporant this period is in CI rules

if we use colors, the player would have to estimtae how long til it reaches green/guess what shade of green is actually the end shade

also, im juggling like 4 different projects right now _and_ i really really hate working in ci code, i COULD do it but i really dont want to, if someone else wants theyre free to do so and i encourage you

**Comment by vinylspiders** [comment]
> id argue its important to maintain the granularity of being able to see the exact numbers til the timeout period expires considering how imporant this period is in CI rules
> 
> if we use colors, the player would have to estimtae how long til it reaches green/guess what shade of green is actually the end shade
> 
> also, im juggling like 4 different projects right now _and_ i really really hate working in ci code, i COULD do it but i really dont want to, if someone else wants theyre free to do so and i encourage you

I get you, but the floating numbers really do look terrible. All you have to do is make the icon gradually become grey, and it would convey the same message essentially.

Even a progressbar would be preferable to this IMO. The numbers are not easy to read, especially while moving, and are visual clutter.

**Comment by YakumoChen** [comment]
Is there sanity checking for if the config is commented out or set to 0 or a negative for whatever reason?

File: config/skyrat/skyrat_config.txt [hunk 1]
```diff
@@ -66,6 +66,9 @@ BLACKOUTPOLICY You remember nothing after you've blacked out and you do not reme
 ## Combat indicator, comment out to disable it
 COMBAT_INDICATOR
```
Comment (YakumoChen) [inline config/skyrat/skyrat_config.txt:68]: Elaborate the number is measured in seconds please

**Comment by Iamgoofball** [comment]
this needs to be like a progress bar looking visual or something

**Comment by burgerenergy** [comment]
In contrast, I think it needs to not have any bar at all, just something that goes to HOLD UP to READY so people can't preempt it.

That said, while we're here; CI has a he-said-she-said issue; would you be willing to make an event in the combat log when the timer/whatever expires? Easier to look up if someone did jump the gun or not for the cops.

**Pull request closed without merging**